src/shared/components/palette/case-link/write-case-link-field.component.ts

```diff
@@ -28,6 +28,14 @@ export class WriteCaseLinkFieldComponent extends AbstractFieldWriteComponent imp
     }
     this.caseReferenceControl = this.caseLinkGroup.controls['CaseReference'];
     this.caseReferenceControl.setValidators(this.caseReferenceValidator());
+
+    // Ensure that all sub-fields inherit the same value for retain_hidden_value as this parent; although a CaseLink
+    // field uses the Complex type, it is meant to be treated as one field
+    if (this.caseField) {
```

Comment:
Should this also check that it's a `Complex` field, considering what it does within the condition?

Reply:
Hmm - that's a good point. `caseField.field_type` is a mandatory property, so it's worth checking this is equal to `Complex`. There is also the same thing in `write-organisation-field.component.ts`, so I'll add it there as well.